perf/src/deduper.rs

```diff
@@ -87,15 +88,19 @@ fn new_random_state<R: Rng>(rng: &mut R) -> RandomState {
     RandomState::with_seeds(rng.gen(), rng.gen(), rng.gen(), rng.gen())
 }
 
+/// Returns the tuple of (total discarded packets, forwarded_packets, forwarded discarded packets)
 pub fn dedup_packets_and_count_discards<const K: usize>(
     deduper: &Deduper<K, [u8]>,
     batches: &mut [PacketBatch],
     mut process_received_packet: impl FnMut(&mut Packet, bool, bool),
-) -> u64 {
+) -> (usize, usize, usize) {
+    let mut total_discards: usize = 0;
+    let mut forwarded_packets: usize = 0;
```

Comment:
Ideologically, I don't like the idea of counting forwards in `Deduper` since `Deduper` shouldn't care about the origin of packets. However, I see that this is a good place to plugin given that we're already iterating through the packets. I guess the deduper is already fairly tied to packet with `.discard()` so I don't think it is a showstopper to me.

Reply:
Sounds good on just using the patch unmerged to collect data for now. I believe you got access to our staked testnet node, but another idea is if you make your patches available, you could ask for volunteers to apply + run with your patch (on testnet and/or mnb)